### DISCUSSION TITLE
Course Pre-Requirements

### DISCUSSION THREAD

**Discussion by lucawrabetz** [huggingface/smol-course#14] [General]
Hi,

First of all thanks for posting this course, I'm really excited to find time to follow along this December and January. Since I was wondering about "pre-requirements", I figured that any of the HF folks' comments on the question would be useful to anyone finding the repo, so I opened a new discussion.

I'm wondering if you've designed this course to be "self-contained", or if some familiarity with language models is expected (if so, how much, what type). The reason I ask is I'm a good example of "not sure where I stand" - I feel very good about applied mathematics and python due to my work, but I only know about LLMs from "following the hype" (though this includes an occasional seminal transformer paper and repeated views of 3Blue1Brown videos). 

Again, this question is not pressing--I will certainly jump in to the first few modules regardless of the responses :). That being said, any comment on this distinction would be super helpful and might lead to more good resources for everyone.

Luca

[Edit: I know there is a prerequisite section in the README, still figured it would be useful to have a thread for pre-requisite related discussions for people who need to review anything.]

**Comment by turbosnail11** [comment]
> [Edit: I know there is a prerequisite section in the README, still figured it would be useful to have a thread for pre-requisite related discussions for people who need to review anything.]

As per the readme - _Familiarity with Python, **PyTorch, and the transformers library**_ -  I think if we were familiar with those we wouldn't need this course?
